[TEST] Kill remaining lang-groovy messy tests

tlrx opened the pull request:
After #13834 many tests that used Groovy scripts (for good or bad reason) in their tests have been moved in the lang-groovy module and the issue #13837 has been created to track these messy tests in order to clean them up.
    
The work started with #19280, #19302 and #19336 and this PR moves the remaining messy tests back in core, removes the dependency on Groovy, changes the scripts in order to use the mocked script engine, and change the tests  to integration tests.

It also moves `IndexLookupIT` test back (even if it has good chance to be removed soon) and fixes its tests.

closes #13837

nik9000 commented on core/src/test/java/org/elasticsearch/script/IndexLookupIT.java:
```diff
@@ -0,0 +1,1016 @@
+            scripts.put("Sum the payloads of [float_payload_field][b]", vars -> payloadSum(vars, "float_payload_field", "b"));
+            scripts.put("Sum the payloads of [int_payload_field][b]", vars -> payloadSum(vars, "int_payload_field", "b"));
+
+            scripts.put("createPositionsArrayScriptIterateTwice[b,_FREQUENCIES|_OFFSETS|_PAYLOADS|_POSITIONS|_CACHE,position]",
```
Maybe `s/_FREQUENCIES|_OFFSETS|_PAYLOADS|_POSITIONS|_CACHE/" + INCLUDE_ALL + "/`?

nik9000 commented on core/src/test/java/org/elasticsearch/script/IndexLookupIT.java:
```diff
@@ -0,0 +1,1016 @@
+
+public class IndexLookupIT extends ESIntegTestCase {
+
+    private static final String includeAllFlag = "_FREQUENCIES|_OFFSETS|_PAYLOADS|_POSITIONS|_CACHE";
```
Can you name it in SHOUTING_SNAKE_CASE like other constants?

tlrx replied:
Sure. I also renamed the constants.

nik9000 commented on core/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricIT.java:
```diff
@@ -748,7 +929,7 @@ public void testEmptyAggregation() throws Exception {
         assertThat(scriptedMetric.aggregation(), instanceOf(List.class));
         @SuppressWarnings("unchecked") // We'll just get a ClassCastException a couple lines down if we're wrong, its ok.
-        List<Integer> aggregationResult = (List<Integer>) scriptedMetric.aggregation();
+                List<Integer> aggregationResult = (List<Integer>) scriptedMetric.aggregation();
```
Extra

nik9000 commented on core/src/test/java/org/elasticsearch/search/sort/SimpleSortIT.java:
```diff
@@ -0,0 +1,470 @@
+        assertAcked(prepareCreate("test")
+                .addMapping("type1", XContentFactory.jsonBuilder().startObject()
+                        .startObject("type1")
+                        .startObject("properties")
```
I think the old indentation was a bit better than this.

nik9000 commented on core/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricIT.java:
```diff
@@ -102,42 +224,65 @@ public void setupSuiteScopeCluster() throws Exception {
+
+            // When using the MockScriptPlugin we can map File scripts to inline scripts:
+            // the name of the file script is used in test method while the source of the file script
+            // must match a predefined script from CustomScriptPlugin.pluginScripts() method
```
I can't find the assertion that matches the comment. Can you walk me through the logic?

tlrx replied:
I added some comments in `MockScriptEngine`, I hope it helps to better understand how it resolves the inline/stored/file scripts.

Please let me know if it's still not clear

nik9000 replied:
It is much more clear to me now. Do we still need to make the files though?

tlrx replied:
Yes. I guess we could mock the file loading too  so that we don't need to create files but I like the fact that `MockeScriptEngine` uses the default stuff and only differs when it comes to script compilation and execution.

nik9000 replied:
Now I get it! I had to dig around a bit in `ScriptService` to understand what was up. Now it makes sense.

tlrx commented:
@nik9000 Thanks for your review!

I addressed all your comments. I also made the `MockScriptEngine` stricter so that it now throws an exception when compiling undefined scripts.

Can you please have another look?

tlrx commented:
@nik9000 Do you need any more information or do you have any more comment? I'd like to get this in. Thanks

nik9000 commented:
LGTM. Sorry to wedge this open so long.

Elasticmachine is complaining about:
```
  2> REPRODUCE WITH: gradle :core:integTest -Dtests.seed=FBA82156698EB076 -Dtests.class=org.elasticsearch.script.IndexLookupIT -Dtests.method="testCallWithDifferentFlagsFails" -Dtests.security.manager=true -Dtests.locale=mk -Dtests.timezone=America/Argentina/San_Juan
```

so I guess that is all that is left to solve before merging.

tlrx commented:
@nik9000 Thanks! I thought I fixed this test but obviously not. It should be OK now.

tlrx commented:
:(

nik9000 commented:
Sad jenkins.

nik9000 commented:
Lots of fun failures this time!

```
   > Caused by: java.lang.IllegalArgumentException: No pre defined script matching [{ "match_all" : {}}] for script with name [null], did you declare the mocked script?
```

tlrx commented:
Yes, I made MockScriptEngine more strict and it now shouts when a script is used but not registered first. Looks like `TemplateQueryBuilderTests` uses mocked scripts and mustache scripts but the way the test is initialized only MockScriptEngine is used. It's almost a useless test.

nik9000 commented:
Last change makes sense to me. LGTM!
